### DISCUSSION TITLE
SCtransform and differential expression in v4

### DISCUSSION THREAD

**Discussion by sanachintamen** [satijalab/seurat#4032] [Q&A] [answered]
Hi @satijalab ,
Thanks for your wonderful work! 

I've been using SCtransform to normalize my datasets as they span different experimental conditions. I had read numerous discussions on which assay and slot to use and I wanted to ask whether there have been updates to the following:

"in principle, it would be most optimal to perform these calculations directly on the residuals (stored in the scale.data slot) themselves. This is not currently supported in Seurat v3, but will be soon." 

Does your group have an updated recommendation on how to perform DGE analysis with SCtransform in the latest version of seurat?

Thanks,
Sana

**Comment by satijalab** [comment, marked as answer]
Thanks for asking. We had anticipated extending Seurat to actively support DE using the pearson residuals of sctransform, but have decided not to do so. In some cases, Pearson residuals may not be directly comparable across different datasets, particularly if there are batch effects that are unrelated to sequencing depth. While it is possible to correct these differences using the SCTransform-based integration workflow for the purposes of visualization/clustering/etc., we do not recommend running differential expression directly on Pearson residuals. Instead, we recommend running DE on the standard RNA assay.

**Reply by duocang** [reply]
HI.

1. So if we run DE on `RNA` assay, should I use `NornalizeData` and `ScaleData` even after `SCTransform` in the very beginning or should we abandon `SCTransfrom` if DE is going to happen? Just like [option 2 from issue 1501](https://github.com/satijalab/seurat/issues/1501).
2. For `RNA` assay, should I use `scale.data` or `data`?

**Reply by jiangshan529** [reply]
> HI.
> 
> 1. So if we run DE on `RNA` assay, should I use `NornalizeData` and `ScaleData` even after `SCTransform` in the very beginning or should we abandon `SCTransfrom` if DE is going to happen? Just like [option 2 from issue 1501](https://github.com/satijalab/seurat/issues/1501).
> 2. For `RNA` assay, should I use `scale.data` or `data`?

Good question. That's also what I am concerning about. Are there answers to the two questions?

**Reply by MrModenait** [reply]
Hi @duocang and @jiangshan529,
While I am no authority on this matter, and it's been some time since I last worked with this data, I can try to explain what I have done:

1. Prior to integrating my datasets, I run both `NormalizeData` and `SCTransform`. I run `NormalizeData` on the counts slot of the `RNA` assay, which created the data slot for this assay. I also run `SCTransform` on the counts slot of the `RNA` assay, creating three slots in the `SCT` assay (`counts`, `data` and `scale.data`). There are two reasons as to why I run `NormalizeData` even though `SCTransform` has existed for some time now: firstly, I need the normalised counts to calculate the cell cycle scores with `CellCycleScoring` that later I regress out in `SCTransform` and secondly, I want to have my normalised raw counts for DE expression later on when I have integrated the dataset. I can see the point of running `NormalizeData` again on the integrated dataset before running DE analysis so that the counts would be normalised across the whole integrated dataset, but this is not something I have considered doing until now.
2. I use the data slot from the `RNA` assay for differential expression analysis and making my violin plots, but I use the `scale.data` slot for (and only for) making my heatmaps with `DoHeatmap`.

**Comment by haoyanioz** [comment]
@satijalab @troels @sanachintamen @mxposed 
Dear Seurat team,
Thank you for your great work about Seurat. Here we have some questions about this software.

Is it suitable to use the slot "data" of SCT assay for running FindAllMarkers and FindMarkers?
For the slot "data" of RNA assay, SCT assay, and integrated assay, which one should we select to perform visualizations of marker feature expression (FeaturePlot, RidgePlot, VlnPlot, and DotPlot)?


Looking forward to your reply. Any help would be highly appreciated. Thanks.

Best~
STAR

**Reply by eiden309** [reply]
@satijalab I have the same question in mind:
If I switch back to the "RNA" assay after running SCTrasnform earlier, I assume I would have to run NormalizeData, FindVariableFeatures, ScaleData on the assay to generate the input data for `FindAllMarkers`.

By default, `FindAllMarkers` uses the slot "data", so if we set `DefaultAssay(clustered_seurat) <- "SCT"`, the log1p(corrected counts) are used as input to `FindAllMarkers`. Since I am using SCTransform instead of the NormalizeData, FindVariableFeatures, ScaleData workflow, would it not be more consistent/appropriate to use the "SCT" assay data slot for finding marker genes?

**Reply by duocang** [reply]
`data` of `SCT` and `RNA` is kinda the same thing, it is believed `SCT` does a "better" job. You can try `SCT`for DE but my experience was that the results are subtle. Since satijalab has said in many issues that `data` of `RNA` is recommended you think you better use it.

**Reply by eiden309** [reply]
@duocang Does that mean that even when you run `SCTransform` for normalization initially, you would run `NormalizeData` and `ScaleData` subsequently on the RNA assay for DE analysis? While I understand that @satijalab has recommended the usage of RNA assay in many issues, I was just puzzled by the fact that we would ultimately need to run all these functions (`SCTransform`, `NormalizeData` and `ScaleData`) in a single pipeline - even though `SCTransform` is meant to replace the other 2 functions.

**Reply by duocang** [reply]
@satijalab once answered the same issue that you @eiden309 asked. [here](https://github.com/satijalab/seurat/issues/1501)
In the above-mentioned link, @satijalab was up to use `obj[["SCT"]]@scale.data` for DE, but `obj[["RNA"]]@data` is recommended.
> Thanks for asking. We had anticipated extending Seurat to actively support DE using the pearson residuals of sctransform, but have decided not to do so. In some cases, Pearson residuals may not be directly comparable across different datasets, particularly if there are batch effects that are unrelated to sequencing depth. While it is possible to correct these differences using the SCTransform-based integration workflow for the purposes of visualization/clustering/etc., we do not recommend running differential expression directly on Pearson residuals. Instead, we recommend running DE on the standard RNA assay.

I feel puzzled too and I am not yet confident to say which option is better or more normal/standard to go.

The easiest way is to stick to `NormalizeData` and `ScaleData` which is still used in recent vignettes (V4) so I think it is not bad to use them.

What I am doing is to rerun  `NormalizeData` and `ScaleData`  even my data has been processed by `SCTransform`. At leaset `RNA` and `SCT` do not effct each other.

**Reply by eiden309** [reply]
I see, thanks for your input @duocang! Hopefully, @satijalab can make it clearer in their future vignettes as this issue has been raised many times

**Reply by MrModenait** [reply]
Hello and sorry for reopening this two-month old thread, but I am struggling to understand which assay I should for plotting my DE results. After one-and-a-half years of working with Seurat, we can all agree on that _differential gene expression analysis is done on the **RNA** assay_. However, how should I interpret my findings when a gene that is downregulated in the RNA assay (and therefore downregulated in my DE results) is in fact _upregulated_ in the SCT data slot, and as a result "brighter" on my heatmap?
To give some numbers, in the `[["RNA"]]@data` slot, a gene has an average expression level of `3.79` for `control` and `1.48` for `disease` (thus downregulated in disease), as calculated by `AverageExpression()`. However, in the `[["SCT"]]@scale.data` slot, which I have used to build my heatmap, the gene has an expression value of `-9.19e-17` in `control` and `-2.71e-20` for `disease`. The SCT-value for `disease` is greater than the value for `control` on the number line, contradicting the `[["RNA"]]@data` results and my DE results. How should I interpret this? Should I perhaps use another slot for my heatmap?

**Comment by Susuqu** [comment]
Hi @MrModenait,  I also got the same different "direction" results as you mentioned, so how could you explain the downregulated and upregulated DEGs? Or any suggestions? Thanks a lot.

> Hello and sorry for reopening this two-month old thread, but I am struggling to understand which assay I should for plotting my DE results. After one-and-a-half years of working with Seurat, we can all agree on that _differential gene expression analysis is done on the **RNA** assay_. However, how should I interpret my findings when a gene that is downregulated in the RNA assay (and therefore downregulated in my DE results) is in fact _upregulated_ in the SCT data slot, and as a result "brighter" on my heatmap? To give some numbers, in the `[["RNA"]]@data` slot, a gene has an average expression level of `3.79` for `control` and `1.48` for `disease` (thus downregulated in disease), as calculated by `AverageExpression()`. However, in the `[["SCT"]]@scale.data` slot, which I have used to build my heatmap, the gene has an expression value of `-9.19e-17` in `control` and `-2.71e-20` for `disease`. The SCT-value for `disease` is greater than the value for `control` on the number line, contradicting the `[["RNA"]]@data` results and my DE results. How should I interpret this? Should I perhaps use another slot for my heatmap?

**Reply by MrModenait** [reply]
Dear @Susuqu,
Glad to see you're interested in the question. My research is on hold for the moment while I'm continuing on my MSc degree so I it's been a while since I looked at my data. The only hypothesis I can offer is that this is the result of the `SCTransform` algorithm correcting the data?
I would still use the `RNA` assay though, since it feels more meaningful to draw conclusions based on actual data than "virtual" data. In general I have used the `SCT` slot for heatmaps and as a bridge to data integration, and I have continued to use `RNA` for all DE analysis.

**Comment by flde** [comment]
I think the Seurat team address many issues mentioned above with SCTransform v2 https://satijalab.org/seurat/articles/sctransform_v2_vignette.html and https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02584-9.
